apps/language_server/lib/language_server/providers/completion.ex

```diff
@@ -587,6 +587,16 @@ defmodule ElixirLS.LanguageServer.Providers.Completion do
     "defmodule #{suggest_module_name(file_path)}$1 do\n\t$0\nend"
   end
 
+  defp snippet_for({"Kernel", "defprotocol"}, %{file_path: file_path})
+       when is_binary(file_path) do
+    "defprotocol #{suggest_module_name(file_path)}$1 do\n\t$0\nend"
+  end
+
+  defp snippet_for({"Kernel", "defimpl"}, %{file_path: file_path})
+       when is_binary(file_path) do
+    "defimpl #{suggest_module_name(file_path)}$1, for: $2 do\n\t$0\nend"
```

Comment:
This won’t likely be a good module name. The protocol is often from some other lib. How about suggesting ‘for’ instead?

Reply:
OK, I got it.

> How about suggesting ‘for’ instead?

If we use `defprotocol` and `defimpl` in one file like `Enumerable`, it doesn't make sense.
So, I delete the `defimpl` pattern match? Or do you have any suggestions?